pkg/client/clientset.go

```diff
@@ -98,17 +109,33 @@ func CreatePlatformClientSet(url string, auth PlatformAuth, opts ClientOptions)
 		ClientSecret: auth.OauthClientSecret,
 		TokenURL:     auth.OauthTokenURL,
 	}
+
+	tokenClient := clientAuth.NewTokenAuthClient(auth.Token)
+	oauthClient := clientAuth.NewOAuthClient(context.TODO(), oauthCredentials)
+	classicURL, err := metadata.GetDynatraceClassicURL(context.TODO(), rest.NewRestClient(oauthClient, nil, rest.CreateRateLimitStrategy()), url) //this will send the default user-agent
```

Comment:
nothing for this PR, just something I realized while reading this: we could move the optional user-agent down to rest client initialization now that we explicitly pass it in to the higher-level-logic `clients`.